imap/smtpclient.c

```diff
@@ -863,6 +891,19 @@ EXPORTED int smtp_is_valid_esmtp_value(const char *val)
     return 1;
 }
 
+/* Encodes into the current location of the struct buf.
+   The caller must buf_reset() before calling is necessary. */
```

Comment:
Nit type: 'is' should mean 'if'. I am generally bad with comma, but I think the "if necessary" should be preceded by one.